### ISSUE TITLE
[BUG] discard action does not work

### ISSUE THREAD

**Issue by GeorgFleig** [rspamd/rspamd#2875]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Read about bug reporting in general: https://rspamd.com/doc/faq.html#how-to-report-bugs-found-in-rspamd
    * Enabled relevant debugging logs: https://rspamd.com/doc/faq.html#how-to-debug-some-module-in-rspamd 
    * Checked the FAQs about Core files in case of fatal crash: https://rspamd.com/doc/faq.html#how-to-figure-out-why-rspamd-process-crashed
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+user%3Arspamd
    * Checked that there is not already an experimental package or master branch

**Describe the bug**
I want to set up user settings to silently discard (not reject) emails from a certain sender. Therefore I configured matching user settings and set the threshold for the "discard" action to 1. The emails in question always have a score above 1, so the threshold should kick in. But rspamd simply ignores the defined threshold for the action. The "discard" action is supposed to be available since 1.9 according to https://rspamd.com/doc/faq.html#what-are-rspamd-actions .

### Steps to Reproduce

1. Create user setting to match certain incoming mails
2. Set threshold for "discard" action to 1 for the matches
3. Receive an email that matches the user setting and will have a score above 1

**Expected behavior**
The email should get silently discarded. Instead nothing happens/the next higher threshold action kicks in. When I set the threshold for the "reject" action instead to 1, the mail gets rejected. This indicates that the matcher for the mail works as expected.

Setting the "discard" threshold for all mails directly in `actions.conf` also does not have any effect.

### Versions

- Rspamd daemon version 1.9.2
- 4.9.0-8-amd64
- Debian Stretch

### Additional Information

`local.d/settings.conf`

```
discard_bad_mails {
  id = "discard_bad_mails";
  priority = "high";
  from = "user@example.org";
  rcpt = "rcpt@example.org";
  apply {
    actions {
      discard = 1.0;
    }
  }
}
```

A mailing list thread seems to be related to this issue: https://lists.rspamd.com/pipermail/users/2019-March/000409.html . Unfortunately without any useful answer. In another message the author indicates that discarding only works with `discard_on_reject = true;` (https://lists.rspamd.com/pipermail/users/2019-March/000429.html).

**Comment by vstakhov** [comment]
You need to define this action first. Not a bug, closing as inappropriate for issue tracker.

**Comment by GeorgFleig** [comment]
Thanks for your feedback. I played around a little more and finally got it to work. Defining the action with a threshold in `local.d/actions.conf` and then setting the threshold in `local.d/settings.conf` for my matcher worked.

I did try that before opening the bug. Turned out the test mail I sent did not reach the threshold of 1 in this case, my mistake. But I also ran `rspamadm configtest`, which yields `unknown element in actions section: discard`. That implied to me that defining the action is wrong, so I opened this bug ticket. Seems like the config validation is not prepared yet for the new and custom actions (see https://github.com/rspamd/rspamd/blob/8920c4d8faf24ab5eff34f389d01da7635514fb0/lualib/lua_cfg_transform.lua#L257)

In addition, I understood the documentation in a way that the new actions are ready to use and don't ned additional defining:
> From version 1.9, there are also some more actions:
> quarantine: push a message to quarantine (must be supported by MTA)
> discard: silently discard a message



Works like a charm now. Thanks for your support and for providing rspamd!
